app/views/articles/_single_story.html.erb

```diff
@@ -162,21 +162,24 @@
           <small class="crayons-story__tertiary fs-xs mr-2">
             <%= t("views.articles.reading_time", count: story.reading_time < 1 ? 1 : story.reading_time) %>
           </small>
-          <button
-            type="button"
-            id="article-save-button-<%= story.id %>"
-            class="c-btn c-btn--icon-alone bookmark-button"
-            data-reactable-id="<%= story.id %>"
-            data-article-author-id="<%= story.user_id %>"
-            aria-label="<%= t("views.articles.save.aria_label", title: story.title) %>"
-            title="<%= t("views.articles.save.title", title: story.title) %>">
-            <span class="bm-initial">
-              <%= inline_svg_tag("small-save.svg", alt: "", aria_hidden: true) %>
-            </span>
-            <span class="bm-success">
-              <%= inline_svg_tag("small-save-filled.svg", alt: "", aria_hidden: true) %>
-            </span>
-          </button>
+
+          <% if (@user == nil) || (@user.id != story.user_id) %>
```

Comment:
Could you give me a bit of context on why we have this if statement? Why do we specifically check if a user is nil? Under what circumstances do we think that the user would  be nil?

For `@user.id != story.user_id)` is this because we only want to show a button on the articles that were not authored by the user?

Reply:
`(@user.id != story.user_id)` -> For this you are correct that we only want to show bookmark icon if article is from a different author.

`(@user == nil)` -> This i added later on when `11 CI checks` failed and some of them mentioned the error that unable to find/use `id` on nil object and a result I added this check and it worked. 
Let me know what can be done better here.

Also, won't the user will be nil for non-signed in users?

Reply:
> Also, won't the user will be nil for non-signed in users?

Yeah this makes sense, I was thinking that the dashboard cannot be accessed by non signed in users but I noticed that the partial is being called from a couple of places.